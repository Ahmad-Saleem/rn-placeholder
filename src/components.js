import { View } from 'react-native';
import connect from './placeholderContainer';
import stylify from './placeholderStylify';
import computeStyleMedia from './media/media.style';
import computeStyleBox from './box/box.style';

const compose = (f, g) => x => f(g(x));

/**
 * Export shared elements
 */
const Media = compose(
  connect,
  stylify(computeStyleMedia),
)(View);

const Box = compose(
  connect,
  stylify(computeStyleBox),
)(View);

export { Media, Box };
