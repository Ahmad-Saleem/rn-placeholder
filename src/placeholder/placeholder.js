import React from 'react';
import { View } from 'react-native';
import Animations from '../animation/animations';

const makeRoot = (animation) => {
  if (animation) {
    const Animation = Animations[animation];

    if (!Animation) {
      throw new Error(`Animation "${animation}" doesn\'t exist in the library`);
    }

    return Animation;
  }

  return View;
};

const Placeholder = ({
  isReady, animation, customAnimation, children, ...props
}) => {
  const Root = customAnimation || makeRoot(animation);

  return (
    isReady && (
      <Root {...props}>
        {React.Children.toArray(children).map(element => React.cloneElement(element))}
      </Root>
    )
  );
};

export default Placeholder;
