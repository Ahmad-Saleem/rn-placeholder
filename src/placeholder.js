import connect from './placeholderContainer';
import Paragraph from './paragraph/paragraph';
import ImageContent from './imageContent/imageContent';
import MultiWords from './multiWords/multiWords';
import Placeholder from './placeholder/placeholder';
import { Media, Line, Box } from './components';

/**
 * Export the placeholder
 */
export default {
  Placeholder,
  ImageContent: connect(ImageContent),
  Paragraph: connect(Paragraph),
  MultiWords: connect(MultiWords),
  Media,
  Line,
  Box,
  connect,
};
