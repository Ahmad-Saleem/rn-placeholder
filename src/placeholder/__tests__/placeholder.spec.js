import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import { Line } from '../../components';
import { Placeholder } from '../placeholder';

describe('placeholder', () => {
  let props;

  const getWrapper = () => shallow(<Placeholder {...props} />);

  beforeEach(() => {
    props = {
      isReady: true,
      children: [<Line />, <Line />, <Line />],
    };
  });

  it('should match snapshot', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('shouldnt display anything when the isReady prop is false', () => {
    props.isReady = false;
    expect(getWrapper()).toMatchSnapshot();
  });

  it('should be wrapped with an Animation when animate prop is se to "fade"', () => {
    props.isReady = false;
    props.animation = 'fade';

    expect(getWrapper()).toMatchSnapshot();
  });

  it('should be wrapped with an Animation when animate prop is se to "fade"', () => {
    props.isReady = false;
    props.animation = 'fade';

    expect(getWrapper()).toMatchSnapshot();
  });

  it('should throw an error when the animation name is not found', () => {
    props.isReady = false;
    props.animation = 'toto';

    expect(() => getWrapper()).toThrow('Animation "toto" doesn\'t exist in the library');
  });

  it('should be wrapped with a CustomAnimation when the customAnimation prop is set', () => {
    props.customAnimation = componentProps => (
      <View style={{ backgroundColor: 'yellow' }} {...componentProps} />
    );

    expect(getWrapper()).toMatchSnapshot();
  });
});
