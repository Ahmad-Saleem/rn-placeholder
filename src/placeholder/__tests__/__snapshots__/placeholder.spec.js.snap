// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder should be wrapped with a CustomAnimation when the customAnimation prop is set 1`] = `
<Component>
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".0"
    onReady={false}
  />
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".1"
    onReady={false}
  />
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".2"
    onReady={false}
  />
</Component>
`;

exports[`placeholder should be wrapped with an Animation when animate prop is set to "fade" 1`] = `
<Fade>
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".0"
    onReady={false}
  />
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".1"
    onReady={false}
  />
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".2"
    onReady={false}
  />
</Fade>
`;

exports[`placeholder should match snapshot 1`] = `
<View>
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".0"
    onReady={false}
  />
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".1"
    onReady={false}
  />
  <placeHolder
    animate={null}
    customAnimate={null}
    key=".2"
    onReady={false}
  />
</View>
`;

exports[`placeholder shouldnt display nothing when the isReady prop is false 1`] = `""`;
